core/modules/system/system.admin.inc

```diff
@@ -519,12 +519,29 @@ function system_modules($form, $form_state = array()) {
   // Get current list of modules.
   $files = system_rebuild_module_data();
 
-  // Remove hidden modules from display list.
   $visible_files = $files;
   foreach ($visible_files as $filename => $file) {
+    // Remove hidden and required modules from display list.
     if (!empty($file->info['hidden'])) {
       unset($visible_files[$filename]);
     }
+    else {
```

Comment:
This entire block seems unnecessary; the variables `$module_tags` and `$all_tags` are not used elsewhere. The actual tags are built in `_system_modules_build_row()`.

Reply:
Yep, I think that was left over from one of the other features. Removing.